src/Lean/PrettyPrinter/Delaborator/Basic.lean

```diff
@@ -251,7 +281,7 @@ partial def delab : Delab := do
     else
       return ← annotateTermInfo (← ``(_))
```

Comment:
If we wanted to adjust `pp.proofs` to use this new syntax, I suppose this `if`/`else` could be replaced with something like
```lean
    let pf ← omission
    if ← getPPOption getPPProofsWithType then
      let stx ← withType delab
      return ← `(($pf : $stx)) >>= annotateCurPos
    else
      return pf
```
(That's untested -- I'm just wanting to record how big of a change this would be.)